src/modules/logicModule/LM_PC_PaymentRouter_v1.sol

```diff
@@ -61,11 +61,18 @@ contract LM_PC_PaymentRouter_v1 is
         address recipient,
         address paymentToken,
         uint amount,
-        uint dueTo
+        uint start,
+        uint cliff,
+        uint end
     ) public onlyModuleRole(PAYMENT_PUSHER_ROLE) {
         _addPaymentOrder(
             PaymentOrder(
-                recipient, paymentToken, amount, block.timestamp, dueTo
+                recipient,
+                paymentToken,
+                amount,
+                start == 0 ? block.timestamp : start,
+                cliff,
```

Comment:
The cliff is being checked in the `StreamingPaymentProcessor`, in the sense that `start + cliff <= end` is required. If `cliff > end - start` it would violate that and throw an error. Or do you propose to already check this here as well?

Reply:
The problem is that in the case that the paymentOrders arent immediatley processed, then we basically create faulty paymentOrders, that cant be processed later on
Id rather check 2 times

Reply:
The paymentRouter _does_ immediately process them though, so we would be double-checking. And the other modules don't implement cliffs. I think we can keep it like that

Reply:
I see your point @FHieser, but also agree with @0xNuggan - we can keep it like this for now I think.

Reply:
We dont enforce this though.
If someone else comes up and wants to develop something this has to be enforced
I will create a linear issue for it so we keep it in mind